src/nevermined/api/AssetsApi.ts

```diff
@@ -0,0 +1,464 @@
+            consumerAccount,
+            files,
+            resultPath,
+            fileIndex
+        )
+        this.logger.log('Files consumed')
+
+        if (resultPath) {
+            return resultPath
+        }
+        return true
+    }
+
+
+    /**
+     * Returns the owner of an asset.
+     * @param did - Decentralized ID.
+     * @returns The address of the owner of the asset
+     */
+    public async owner(did: string): Promise<string> {
+        const ddo = await this.resolve(did)
+        const checksum = ddo.checksum(didZeroX(did))
+        const { creator, signatureValue } = ddo.proof
+        const signer = await this.nevermined.utils.signature.verifyText(
+            checksum,
+            signatureValue
+        )
+
+        if (signer.toLowerCase() !== creator.toLowerCase()) {
+            this.logger.warn(
+                `Owner of ${ddo.id} doesn't match. Expected ${creator} instead of ${signer}.`
+            )
+        }
+
+        return creator
+    }
+
+    /**
+     * Returns the assets owned by an address
+     * @param owner - The address to check 
+     * @returns List of DIDs owned by the address
+     */
+    public async ownerAssets(owner: string): Promise<string[]> {
+        return this.nevermined.keeper.didRegistry.getAttributesByOwner(owner)
+    }
+
+    /**
+     * Transfer ownership of an asset.
+     * @param did - Asset DID.
+     * @param newOwner - Ethereum address of the new owner of the DID.
+     * @returns Returns ethers transaction receipt.
+     */
+    public async transferOwnership(
+        did: string,
+        newOwner: string,
+        params?: TxParameters
+    ): Promise<ContractReceipt> {
+        const owner = await this.nevermined.assets.owner(did)
+        return this.nevermined.keeper.didRegistry.transferDIDOwnership(
+            did,
+            newOwner,
+            owner,
+            params
+        )
+    }
+
+    /**
+     * Returns the assets of a consumer.
+     * @param consumer - Consumer address.
+     * @returns List of DIDs.
+     */
+    public async consumerAssets(consumer: string): Promise<string[]> {
+        return (
+            await this.nevermined.keeper.conditions.accessCondition.getGrantedDidByConsumer(
+                consumer
+            )
+        ).map(({ did }) => did)
+    }
+
+
+    public async retire(did: string) {
+        return this.nevermined.services.metadata.delete(did)
+    }
+
+    /**
+     * Download the asset
+     *
+     * @param did - The Decentralized Identifier of the asset.
+     * @param ownerAccount - The receiver account owner
+     * @param resultPath - Path to be the files downloader
+     * @param fileIndex - the index of the file
+     * @param isToDownload - If the NFT is for downloading
+     * @param serviceType - service type. 'access' by default
+     *
+     * @return status, path destination if resultPath is provided or file object if isToDownload is false
+     */
+    public async download(
```

Comment:
Not sure there is a difference between download and consume

Reply:
It allows to download of the files attached to the asset by their owner or provider. This method only can be called successfully by the owner of the asset or a provider.
Added documentation to clarify.